kornia/geometry/bbox.py

```diff
@@ -0,0 +1,500 @@
+    Args:
+        boxes (torch.Tensor): a tensor containing the coordinates of the bounding boxes to be extracted.
+            The tensor must have the shape of Bx8x3, where each box is defined in the following (clockwise)
+            order: front-top-left, front-top-right, front-bottom-right, front-bottom-left, back-top-left,
+            back-top-right, back-bottom-right, back-bottom-left. The coordinates must be in the x, y, z order.
+    """
+    assert len(boxes.shape) == 3 and boxes.shape[1:] == torch.Size(
+        [8, 3]
+    ), f"Box shape must be (B, 8, 3). Got {boxes.shape}."
+
+    left = torch.index_select(boxes, 1, torch.tensor([1, 2, 5, 6], device=boxes.device, dtype=torch.long))[:, :, 0]
+    right = torch.index_select(boxes, 1, torch.tensor([0, 3, 4, 7], device=boxes.device, dtype=torch.long))[:, :, 0]
+    widths = left - right + 1
+    assert torch.allclose(
+        widths.permute(1, 0), widths[:, 0]
+    ), f"Boxes must have be cube, while get different widths {widths}."
+
+    bot = torch.index_select(boxes, 1, torch.tensor([2, 3, 6, 7], device=boxes.device, dtype=torch.long))[:, :, 1]
+    upper = torch.index_select(boxes, 1, torch.tensor([0, 1, 4, 5], device=boxes.device, dtype=torch.long))[:, :, 1]
+    heights = bot - upper + 1
+    assert torch.allclose(
+        heights.permute(1, 0), heights[:, 0]
+    ), f"Boxes must have be cube, while get different heights {heights}."
+
+    depths = boxes[:, 4:, 2] - boxes[:, :4, 2] + 1
+    assert torch.allclose(
+        depths.permute(1, 0), depths[:, 0]
+    ), f"Boxes must have be cube, while get different depths {depths}."
+
+    return True
+
+
+@torch.jit.ignore
+def validate_bbox(boxes: torch.Tensor) -> bool:
+    """Validate if a bounding box is usable or not. BBox can be both 2D or 3D.
+
+    This function checks if the boxes are rectangular or not.
+
+    Args:
+        boxes (torch.Tensor): a tensor containing the coordinates of the bounding boxes to be extracted. The tensor must
+            have the shape of Bx4x2 for 2D bouding box (Bx8x3 in 3D case), where each box is defined in the following
+            (clockwise) order: top-left, top-right, bottom-right, bottom-left and front-top-left, front-top-right,
+            front-bottom-right, front-bottom-left, back-top-left, back-top-right, back-bottom-right, back-bottom-left
+            in 3D case. The coordinates must be in the x, y, (and optionally z) order.
+    """
+
+    assert len(boxes.shape) == 3 and (
+        boxes.shape[1:] == torch.Size([4, 2]) or boxes.shape[1:] == torch.Size([8, 3])
+    ), f"Box shape must be the shape of (B, 4, 2) for 2D bbox or (B, 8, 3) for 3D bbox. Got {boxes.shape}."
+
+    # 2D boxes
+    if boxes.shape[-1] == 2:
+        return validate_bbox_2d(boxes)
+
+    # 3D boxes
+    elif boxes.shape[-1] == 3:
+        return validate_bbox_3d(boxes)
+
+    return True
+
+
+def infer_bbox_shape_2d(boxes: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
```

Comment:
This is the tricky part. I've prepared `validate_bbox` as an example of what I want to do in the future. Adding generic `infer_bbox_shape` doesn't work when wrapping with `jit`, because it returns `Tuple[torch.Tensor, torch.Tensor]` in 2d case and `Tuple[torch.Tensor, torch.Tensor, torch.Tensor]` in 3d case and ASAIK `jit` doesn't support Union so far. I am planning to add a generic function for every bbox-related method, but it will take more time, because I don't know `jit`. Do you want me to do in in this PR?

Reply:
As a dirty workaround we can for now add `@torch.jit.ignore` with `infer_bbox_shape` but allow to wrap it around `infer_bbox_shape_2d` and `infer_bbox_shape_3d`.

Reply:
Ok, generic `infer_bbox_shape` is added, I'll add generic `bbox_to_mask` and `bbox_generator` soon.